### ISSUE TITLE
Can't install a .deb package from the internet using the apt module.

### ISSUE THREAD

**Issue by albrin** [ansible/ansible#29104]
##### ISSUE TYPE

 
 
 - Bug report

##### COMPONENT NAME

apt

##### ANSIBLE VERSION

```
ansible 2.3.2.0
  config file = /vagrant/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]
```

##### CONFIGURATION

default

##### OS / ENVIRONMENT

Ubuntu 17.04

##### SUMMARY

Can not add the MySQL APT Repository.

##### STEPS TO REPRODUCE

```yaml
- name: Install a .deb package from the internet.
  apt:
    deb: https://dev.mysql.com/get/mysql-apt-config_0.8.7-1_all.deb
```

##### EXPECTED RESULTS

The `mysql.list` file will be created in the /etc/apt/sources.list.d/ directory

##### ACTUAL RESULTS

The file `mysql.list` is not created.

```
TASK [albrin.mysql : Install a .deb package from the internet.] *****************************************************************************************************************
task path: /vagrant/ansible/roles/albrin.mysql/tasks/install.yml:50
Using module file /usr/lib/python2.7/dist-packages/ansible/modules/packaging/os/apt.py
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: vagrant
<localhost> EXEC /bin/sh -c 'echo ~ && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/vagrant/.ansible/tmp/ansible-tmp-1504813955.64-196326445886618 `" && echo ansible-tmp-1504813955.64-196326445886618="` echo /home/vagrant/.ansible/tmp/ansible-tmp-1504813955.64-196326445886618 `" ) && sleep 0'
<localhost> PUT /tmp/tmpL0FLKp TO /home/vagrant/.ansible/tmp/ansible-tmp-1504813955.64-196326445886618/apt.py
<localhost> EXEC /bin/sh -c 'chmod u+x /home/vagrant/.ansible/tmp/ansible-tmp-1504813955.64-196326445886618/ /home/vagrant/.ansible/tmp/ansible-tmp-1504813955.64-196326445886618/apt.py && sleep 0'
<localhost> EXEC /bin/sh -c 'sudo -H -S -n -u root /bin/sh -c '"'"'echo BECOME-SUCCESS-icmuvysronoyltznfrsylehwaadxeqmj; /usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1504813955.64-196326445886618/apt.py; rm -rf "/home/vagrant/.ansible/tmp/ansible-tmp-1504813955.64-196326445886618/" > /dev/null 2>&1'"'"' && sleep 0'
ok: [localhost] => {
    "changed": false, 
    "diff": "", 
    "invocation": {
        "module_args": {
            "allow_unauthenticated": false, 
            "autoremove": null, 
            "cache_valid_time": 0, 
            "deb": "/tmp/ansible_iEnaf2/mysql-apt-config_0.8.7-1_all.deb", 
            "default_release": null, 
            "dpkg_options": "force-confdef,force-confold", 
            "force": false, 
            "install_recommends": null, 
            "only_upgrade": false, 
            "package": null, 
            "purge": false, 
            "state": "present", 
            "update_cache": null, 
            "upgrade": null
        }
    }, 
    "stderr": "", 
    "stderr_lines": [], 
    "stdout": "", 
    "stdout_lines": []
}
META: ran handlers
META: ran handlers
```

**Comment by ansibot** [comment]
cc @mgwilliams
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by andreaso** [comment]
Just tested that task myself, using Ubuntu 17.04 and Ansible 2.3.2.0. For me it worked as expected, the page got installed, and _/etc/apt/sources.list.d/mysql.list_ was created.

Noted the following in the output you provided.

    changed": false,

Any chance that the _mysql-apt-config_ had been installed previously?

    $ dpkg --list | grep mysql-apt-config

**Comment by albrin** [comment]
@andreaso, thx.

I tried to run the task of installing the package on a clean image of Ubuntu 17.04 - the file `/etc/apt/sources.list.d/mysql.list` was created.

But if you manually delete the `/etc/apt/sources.list.d/mysql.list` file, when you run the Ansible task again, the file is not created `/etc/apt/sources.list.d/mysql.list`

I do not understand yet what this might be connected with. What does apt look at when it is restarted, from which it can not re-create this file? Is this the correct behavior of Ansible?

P.S.: First time apparently I tried to install the .deb package manually, then I deleted the file `/etc/apt/sources.list.d/mysql.list` and tried to install via Ansible - that's why it did not work either.
